components/notifier/src/destinations/ms_teams.py

```diff
@@ -6,46 +6,43 @@
 
 from models.notification import MetricNotificationData, Notification
 
-
-def notification_text(notification: Notification) -> str:
-    """Create and format the contents of the notification."""
-    nr_changed = len(notification.metrics)
-    plural_s = "s" if nr_changed > 1 else ""
-    markdown = f"{notification.report_title} has {nr_changed} metric{plural_s} that changed status:\n\n"
-    for subject_name in sorted({metric.subject_name for metric in notification.metrics}):
-        markdown += _subject_notification_text(notification, subject_name)
-    return markdown
-
-
-def _subject_notification_text(notification: Notification, subject_name: str) -> str:
-    """Return the notification text for the subject."""
-    markdown = f"* {subject_name}:\n"
-    subject_metrics = [metric for metric in notification.metrics if metric.subject_name == subject_name]
-    for metric in sorted(subject_metrics, key=lambda metric: metric.metric_name):
-        markdown += _metric_notification_text(metric)
-    return markdown
-
-
-def _metric_notification_text(metric: MetricNotificationData) -> str:
-    """Return the notification text for the metric."""
+ICON_URL = "https://raw.githubusercontent.com/ICTU/quality-time/master/resources/icons/%s.png"
+
+
+def metric_section(metric: MetricNotificationData, report_url: str) -> pymsteams.cardsection:
+    """Create a section for a metric status change."""
+    section = pymsteams.cardsection()
+    section.activityTitle(metric.metric_name)
+    section.activitySubtitle(metric.subject_name)
+    section.activityImage(ICON_URL % metric.status)
+    old_status = metric.old_metric_status
+    old_status_text = " (unchanged)" if metric.new_metric_status == old_status else f", was {old_status}"
+    section.addFact("Status:", f"{metric.new_metric_status}{old_status_text}")
+    unit = metric.metric_unit
+    unit_text = unit if (not unit or unit.startswith("%")) else f" {unit}"
```

Comment:
Perhaps it is better to use `unit.isalpha()` instead of starting with %, but this would sadly not match units which contain spaces. Only checking on percent would mismatch a lot of custom units though?

Reply:
% is the only "unit" that does not need a space between the value and the unit, that's why it's treated differently.